### ISSUE TITLE
What / Where is tekDebug?

### ISSUE THREAD

**Issue by b-morgan** [Xruptor/BagSync#201]
As an addon author, I'm always looking for better debugging tools. You mentioned this in your changelog. Where can I find it?

**Comment by Xruptor** [comment]
it's a really old addon used by Tekkub to log debug messages.  That way it doesn't clutter the ChatFrame and can have individual tabs to store the information.  I've used it for ages.

Here is my updated fork for it.
https://github.com/Xruptor/tekdebug

https://github.com/Xruptor/tekdebug/releases/tag/v3.2

Just look at how I setup my Debug() function in BagSync or any other my addons.  You can even put in Module names per debug.  Helps out to track stuff.

**Comment by Xruptor** [comment]
Oh and before I forget @b-morgan :)  Feel free to ask me any questions, or provide insight or suggestions.  I welcome them and appreciate talking to other programmers.  So if you ever see anything wrong as an error or have a better suggestion, feel free to post!  👍

**Comment by b-morgan** [comment]
Thanks for the github pointers. 

I have my own "debug aids" for Skillet (and Skillet-Classic). They can spam chat and/or generate a table of the debug output that is saved in the per character saved variables file. I can have a user that reports a bug that I can't reproduce enable the debugging output using slash commands and then they can send me the saved variables file(s).

I also use a modified version of ViragDevTool.

**Comment by Xruptor** [comment]
> ViragDevTool

You are very welcome!  I do something something similar and request the SavedVariables from users.  It allows me to see their database and load it in a FooBarBagSync simulated one that I created for testing.  I normally use tekDebug to print output information so I can track things down and view variables as they parse.  I don't typically leave the Debugging code in BagSync but as of late I find myself constantly putting it back in.  So I'm just going to hard code it and have a boolean switch to turn it on.

I will admit I've never heard of ViragDevTool.  It looks like an old addon called Lua Browser that I used to use.  It allows you to see functions and variable calls. https://www.curseforge.com/wow/addons/lua-browser  I haven't used it in ages though.  Perhaps I'll give ViragDevTool a look.  Thanks!

**Comment by b-morgan** [comment]
I should probably create a github repository for my version of ViragDevTool. 

There are some changes to the _retail_ (100002) version because of the UI changes and I haven't figured out how to resolve those differences on Wrath and Classic Era.

**Comment by Xruptor** [comment]
> I should probably create a github repository for my version of ViragDevTool.
> 
> There are some changes to the _retail_ (100002) version because of the UI changes and I haven't figured out how to resolve those differences on Wrath and Classic Era.

I took a look at the video for it and it does seem useful and very similar to the old LUA Browser addon I mentioned.  For the Classic and Wrath errors.  Probably best to check for Retail/Classic and then make changes according to that for compatibility.  Since Retail has a lot of changes because of 10.0.2.

Honestly I'm so frustrated with the Debugging stuff I'm just going in and updating it across the board.  Adding debug levels and a bunch of stuff to make it easier for me.  I'm tired of having to manually add the Debug lines LOL.  I should have done this ages ago, but I didn't want to dirty the code with Debug messages everywhere.  At this point I just don't care anymore XD

**Comment by b-morgan** [comment]
Using my DebugAids.lua, the debug statements are DA.DEBUG(...) which I change to --DA.DEBUG(...) so the code still has most of the code but as comments. I can easily mass change "DA.DEBUG" to "--DA.DEBUG" and vice versa. The first parameter to DA.DEBUG is a level so I can control how much chat spam is generated. The saved variables log normally outputs all the debug output along with the level number (but I can also limit that with an option).

**Comment by Xruptor** [comment]
> Using my DebugAids.lua, the debug statements are DA.DEBUG(...) which I change to --DA.DEBUG(...) so the code still has most of the code but as comments. I can easily mass change "DA.DEBUG" to "--DA.DEBUG" and vice versa. The first parameter to DA.DEBUG is a level so I can control how much chat spam is generated. The saved variables log normally outputs all the debug output along with the level number (but I can also limit that with an option). (DA gets assigned to the addon's table so the functions are defined in the addon namespace)

That would be neat :)  I'd love to give the debug a try to test it out.  Perhaps I'll incorporate it in BagSync if it has better control than what I'm used to.

**Comment by Xruptor** [comment]
XD well I meant in terms of your modifications to ViragDevTool.

**Comment by b-morgan** [comment]
DebugAids.lua can be found in the repository for [Skillet](https://github.com/b-morgan/Skillet). It works on all flavors of WoW but will need a slight modification to the definition of DA at the top so that it matches the definition in the addon it is a part of.

I am a novice GitHub user and I probably need some help. I cloned the "official" repository for ViragDevTool to my local machine and started to add my modifications including multiple toc files so it would be useable across all flavors of WoW.
What I should have done is cloned the "official" repository to my GitHub account and then updated it there. I'm not sure how to get from where I am now to where I should be so here's a copy of my current Dragonflight version.

[ViragDevTool.zip](https://github.com/Xruptor/BagSync/files/10098600/ViragDevTool.zip)

Basically, there were a couple of frame resizing issues and I had to change the strata to match the frames I was debugging. What I want to do is add a slash command to change the strata on the fly and fix the frame resize issues.

Hope this is useful.

**Comment by Xruptor** [comment]
Thanks!  I'll take a look :)  That was very kind of you to put it together for me.

**Comment by Xruptor** [comment]
Hey just wanted to let you know I found this fork of the ViragDevTool.  It looks like they even updated it to Ace3 libraries and it was updated 17 days ago.

https://github.com/brittyazel/ViragDevTool

https://github.com/brittyazel/ViragDevTool/archive/refs/heads/main.zip

**Comment by b-morgan** [comment]
Yes, while I was looking around I found the correct fixes for 10.0.0 resizing. Not sure yet if they port back to Wrath (3.4.0).

Its a shame that the author is MIA and there is a license issue. I'd like to see someone take this over and support it.
